static/app/components/smartSearchBar/index.tsx

```diff
@@ -749,13 +769,222 @@ class SmartSearchBar extends React.Component<Props, State> {
+
+    const preparedQuery =
+      typeof prepareQuery === 'function' ? prepareQuery(query) : query;
+
+    // filter existing items immediately, until API can return
+    // with actual tag value results
+    const filteredSearchGroups = !preparedQuery
+      ? this.state.searchGroups
+      : this.state.searchGroups.filter(
+          item => item.value && item.value.indexOf(preparedQuery) !== -1
+        );
+
+    this.setState({
+      searchTerm: query,
+      searchGroups: filteredSearchGroups,
+    });
+
+    const tag = supportedTags[tagName];
+
+    if (!tag) {
+      return {
+        searchItems: [],
+        recentSearchItems: [],
+        tagName,
+        type: 'invalid-tag',
+      };
+    }
+
+    // Ignore the environment tag if the feature is active and
+    // excludeEnvironment = true
+    if (excludeEnvironment && tagName === 'environment') {
+      return null;
+    }
+
+    const fetchTagValuesFn =
+      tag.key === 'firstRelease'
+        ? this.getReleases
+        : tag.predefined
+        ? this.getPredefinedTagValues
+        : this.getTagValues;
+
+    const [tagValues, recentSearches] = await Promise.all([
+      fetchTagValuesFn(tag, preparedQuery),
+      this.getRecentSearches(),
+    ]);
+
+    return {
+      searchItems: tagValues ?? [],
+      recentSearchItems: recentSearches ?? [],
+      tagName: tag.key,
+      type: 'tag-value',
+    };
+  };
+
+  generateOpAutocompleteEntries(
+    validOps: readonly TermOperator[],
+    tag: string
+  ): SearchItem[] {
+    const operatorMap = {
+      [TermOperator.Default]: {
+        type: 'tag-operator' as ItemType,
+        value: ':',
+        desc: t(`${tag}:[value] is equal to`),
+      },
+      [TermOperator.GreaterThanEqual]: {
+        type: 'tag-operator' as ItemType,
+        value: ':>=',
+        desc: t(`${tag}:>=[value] is greater than or equal to`),
+      },
+      [TermOperator.LessThanEqual]: {
+        type: 'tag-operator' as ItemType,
+        value: ':<=',
+        desc: t(`${tag}:<=[value] is less than or equal to`),
+      },
+      [TermOperator.GreaterThan]: {
+        type: 'tag-operator' as ItemType,
+        value: ':>',
+        desc: t(`${tag}:>[value] is greater than`),
+      },
+      [TermOperator.LessThan]: {
+        type: 'tag-operator' as ItemType,
+        value: ':<',
+        desc: t(`${tag}:<[value] is less than`),
+      },
+      [TermOperator.Equal]: {
+        type: 'tag-operator' as ItemType,
+        value: ':=',
+        desc: t(`${tag}:=[value] is equal to`),
+      },
+      [TermOperator.NotEqual]: {
+        type: 'tag-operator' as ItemType,
+        value: '!:',
+        desc: t(`!${tag}:[value] is not equal to`),
+      },
+    };
+    return validOps.map(op => operatorMap[op]);
+  }
+
+  getValidOps(filterToken: TokenResult<Token.Filter>): readonly TermOperator[] {
+    const types = filterToken.invalid?.expectedType ?? [filterToken.filter];
+    const filterType = interchangeableFilterOperators[types[0]]
```

Comment:
It might be more "correct" to map the `types` list here, and not just grab the first one. So we'll end up with a list of token types, and then we pull all the configs out and union them together.

Something like

```tsx
	// If the token is invalid we want to use the possible expected types as our filter type
	const validTypes = filterToken.invalid?.expectedType ?? [filterToken.filter];

	// Determine any interchangable filter types for our valid types
	const interchangableTypes = validTypes.map(
		type => interchangeableFilterOperators[type] ?? []
	);

	// Combine all types
	const allValidTypes = [...new Set(...validTypes, ...interchangableTypes)];
	
	// Find all valid operations
	const validOps = new Set(allValidTypes.map(type => filterTypeConfig[type].validOps).flat())
```